core/vm/evmc.go

```diff
@@ -0,0 +1,395 @@
+		Data:        data,
+		BlockNumber: host.env.BlockNumber.Uint64(),
+	})
+}
+
+func (host *hostContext) Call(kind evmc.CallKind,
+	destination common.Address, sender common.Address, value *big.Int, input []byte, gas int64, depth int,
+	static bool, salt *big.Int) (output []byte, gasLeft int64, createAddr common.Address, err error) {
+
+	gasU := uint64(gas)
+	var gasLeftU uint64
+
+	switch kind {
+	case evmc.Call:
+		if static {
+			output, gasLeftU, err = host.env.StaticCall(host.contract, destination, input, gasU)
+		} else {
+			output, gasLeftU, err = host.env.Call(host.contract, destination, input, gasU, value)
+		}
+	case evmc.DelegateCall:
+		output, gasLeftU, err = host.env.DelegateCall(host.contract, destination, input, gasU)
+	case evmc.CallCode:
+		output, gasLeftU, err = host.env.CallCode(host.contract, destination, input, gasU, value)
+	case evmc.Create:
+		var createOutput []byte
+		createOutput, createAddr, gasLeftU, err = host.env.Create(host.contract, input, gasU, value)
+		isHomestead := host.env.ChainConfig().IsEnabled(host.env.ChainConfig().GetEIP7Transition, host.env.BlockNumber)
+		if !isHomestead && err == ErrCodeStoreOutOfGas {
+			err = nil
+		}
+		if err == errExecutionReverted {
+			// Assign return buffer from REVERT.
+			// TODO: Bad API design: return data buffer and the code is returned in the same place. In worst case
+			//       the code is returned also when there is not enough funds to deploy the code.
+			output = createOutput
+		}
+	case evmc.Create2:
+		var createOutput []byte
+		createOutput, createAddr, gasLeftU, err = host.env.Create2(host.contract, input, gasU, value, salt)
+		if err == errExecutionReverted {
+			// Assign return buffer from REVERT.
+			// TODO: Bad API design: return data buffer and the code is returned in the same place. In worst case
+			//       the code is returned also when there is not enough funds to deploy the code.
+			output = createOutput
+		}
+	default:
+		panic(fmt.Errorf("EVMC: Unknown call kind %d", kind))
+	}
+
+	// Map errors.
+	if err == errExecutionReverted {
+		err = evmc.Revert
+	} else if err != nil {
+		err = evmc.Failure
+	}
+
+	gasLeft = int64(gasLeftU)
+	return output, gasLeft, createAddr, err
+}
+
+// getRevision translates ChainConfig's HF block information into EVMC revision.
+func getRevision(env *EVM) evmc.Revision {
+	n := env.BlockNumber
+	conf := env.ChainConfig()
+	switch {
+	// This is an example of choosing to use an "abstracted" idea
+	// about chain config, where I'm choosing to prioritize "indicative" features
+	// as identifiers for Fork-Feature-Groups. Note that this is very different
+	// than using Feature-complete sets to assert "did Forkage." I may not be
+	// correct here (ie it may not 'correctly' translate to the EVMC logic. This
+	// is why building interoperable tools is annoying; there's no API... yet).
+	// eg. What does your Istanbul mean?
+	case conf.IsEnabled(conf.GetEIP1884Transition, n):
+		return evmc.Istanbul
```

Comment:
We need documentation for this, so user knows that core-geth config can possibly be not 1:1 with EVMC VM, and things might break.